### PR TITLE
Switch pipeline spark sample to hobo compute

### PR DESCRIPTION
**Pull request by YingChen1996**
# Description

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/python/jobs/pipelines/1b_pipeline_with_python_function_components/pipeline_with_python_function_components.ipynb [hunk 1]
```diff
@@ -182,7 +182,7 @@
    "outputs": [],
    "source": [
     "cluster_name = \"cpu-cluster\"\n",
-    "custom_path = 'azureml://datastores/workspaceblobstore/paths/custom_path/${{name}}/'\n",
+    "custom_path = \"azureml://datastores/workspaceblobstore/paths/custom_path/${{name}}/\"\n",
```
Comment (wangchao1230) [inline sdk/python/jobs/pipelines/1b_pipeline_with_python_function_components/pipeline_with_python_function_components.ipynb:185]: why this file changed?

File: sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb [hunk 2]
```diff
@@ -123,9 +123,11 @@
     "    command=\"curl ${{inputs.url}} > ${{outputs.output_folder}}/${{inputs.file_name}}\",\n",
     "    environment=environment,\n",
     "    inputs=dict(url=download_url, file_name=file_name),\n",
-    "    # example how to change path of output on step level, \n",
```
Comment (wangchao1230) [inline sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb:126]: why this file changed?

**Review by wangchao1230 — APPROVED** [review]

File: sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb (hunk 2 above)
Comment (YingChen1996) [reply sdk/python/jobs/pipelines/1d_pipeline_with_non_python_components/pipeline_with_non_python_components.ipynb:126]: smoke/build will conduct black check, and I formatted files detected by black to make smoke/build CI passed. So, this file changed

File: sdk/python/jobs/pipelines/1b_pipeline_with_python_function_components/pipeline_with_python_function_components.ipynb (hunk 1 above)
Comment (YingChen1996) [reply sdk/python/jobs/pipelines/1b_pipeline_with_python_function_components/pipeline_with_python_function_components.ipynb:185]: same with below

**Review by lllilithyang — APPROVED** [review]

**Review by zhengfeiwang — APPROVED** [review]

**Pull request merged**